docs/examples/te_llama/tutorial_accelerate_hf_llama_with_te.ipynb

```diff
@@ -0,0 +1,968 @@
+    "          (down_proj): Linear(in_features=11008, out_features=4096, bias=False)\n",
+    "          (act_fn): SiLU()\n",
+    "        )\n",
+    "        (input_layernorm): LlamaRMSNorm()\n",
+    "        (post_attention_layernorm): LlamaRMSNorm()\n",
+    "      )\n",
+    "    )\n",
+    "    (norm): LlamaRMSNorm()\n",
+    "  )\n",
+    "  (lm_head): Linear(in_features=4096, out_features=32000, bias=False)\n",
+    ")\n",
+    "```\n",
+    "\n",
+    "#### Hugging Face's `LlamaDecoderLayer`\n",
+    "\n",
+    "Let's take a closer look at `LlamaDecoderLayer`. It's composed of `input_layernorm`, `self_attn`, `post_attention_layernorm` and `mlp` modules. Each module has associated weights as shown in the diagram.\n",
+    "\n",
+    "<figure align=\"center\">\n",
+    "<img src=\"media/llama_zoom.png\" width=\"70%\">\n",
+    "    <figcaption> Fig 4: Causal Llama Model Block Diagram (with simplified illustration of the <a href=\"https://github.com/huggingface/transformers/blob/e770f0316d2a9b787c9d1440f204fcb65e176682/src/transformers/models/llama/modeling_llama.py#L695\">LlamaDecoderLayer</a>). </figcaption>\n",
+    "</figure>\n",
+    "\n",
+    "##### Self_Attn Layer\n",
+    "For simplicity in the block diagram illustration of the \"self_attn\" box, we omit the \"Grouped Query Attention\" operation and only showcase the modules which have associated weights.\n",
+    "   \n",
+    "##### MLP Layer\n",
+    "\n",
+    "SwiGLU is an activation defined as follows in the [modeling_llama.py](https://github.com/huggingface/transformers/blob/7c4995f93d8d24aae05e1e43279c96dce736e5c8/src/transformers/models/llama/modeling_llama.py#L236) file in the Hugging Face github repo:\n",
+    "```\n",
+    "down_proj = self.down_proj(self.act_fn(self.gate_proj(x)) * self.up_proj(x))\n",
+    "```\n",
+    "It requires a set of 3 weights as compared to 2 weights in conventional \"MLP\" layers e.g. in the traditional transformer architecture."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2a20adf8",
+   "metadata": {},
+   "source": [
+    "## [Baseline] Running HF `LlamaModel` (Precision: `BF16`)\n",
+    "\n",
+    "Llama2 weights are loaded into the Hugging Face native implementation `LlamaForCausalLM` (refer to [modeling_llama.py](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py)). \n",
+    "\n",
+    "`batch_size` is `8` and precision is `BF16`\n",
+    "\n",
+    "The `LlamaDecoderLayer` is left unchanged in the baseline as follows:\n",
+    "\n",
+    "<figure align=\"center\">\n",
+    "<img src=\"media/llamadecoderlayer.png\" width=\"30%\">\n",
+    "    <figcaption> Fig 5: Revisiting \"LlamaDecoderLayer\". </figcaption>\n",
+    "</figure>\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "f794f2dc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
```

Comment:
Why do you have this cell if you are going to restart the jupyter notebook anyway?